homeassistant/components/tuya/light.py

```diff
@@ -445,6 +453,9 @@ def __init__(
                     self._brightness_type and self._brightness_type.max > 255
                 ):
                     self._color_data_type = DEFAULT_COLOR_TYPE_DATA_V2
+                #If device category is dd (Strip light) use 0-1000 range for colour_data key
+                if (self.device.category == "dd"):
+                    self._color_data_type = DEFAULT_COLOR_TYPE_DATA_1000
```

Comment:
We should avoid doing magic in our code based on the category code.

Instead, let's make sure we can define the entity description

Reply:
Not sure I understand

Reply:
Which part?

Reply:
"define the entity description"

Reply:
See the `LIGHTS` dictionary created in top of this file.

Reply:
I see. How do we check for this description beeing "dd"?

Reply:
Should we use LIGHTS[0]=="dd"?

Reply:
We don't. Instead we should extend the entity description used with a variable defining the fallback color type and it becomes something like this here:

`self._color_data_type = self.entity_description.fallback_color_type`

Reply:
You mean whe shoul add a fallback_color_type here?
```
 # Strip Lights
    # https://developer.tuya.com/en/docs/iot/dd?id=Kaof804aibg2l
    "dd": (
        TuyaLightEntityDescription(
            key=DPCode.SWITCH_LED,
            color_mode=DPCode.WORK_MODE,
            brightness=DPCode.BRIGHT_VALUE,
            color_temp=DPCode.TEMP_VALUE,
            color_data=DPCode.COLOUR_DATA,
        ),
    ),
```

Reply:
yes

Reply:
So something like this?
```
"dd": (
        TuyaLightEntityDescription(
            key=DPCode.SWITCH_LED,
            color_mode=DPCode.WORK_MODE,
            brightness=DPCode.BRIGHT_VALUE,
            color_temp=DPCode.TEMP_VALUE,
            color_data=(DPCode.COLOUR_DATA_V2, DPCode.COLOUR_DATA),
        ),
    ),
```
I thought using DPCode.COLOUR_DATA_V2 meant using colour_data_v2 key and that will not work

Or did you mean like this?
```
"dd": (
        TuyaLightEntityDescription(
            key=DPCode.SWITCH_LED,
            color_mode=DPCode.WORK_MODE,
            brightness=DPCode.BRIGHT_VALUE,
            color_temp=DPCode.TEMP_VALUE,
            color_data=DPCode.COLOUR_DATA,
            fallback_color_type=???
        ),
    ),
```

Reply:
Are you OK with me committing the solution? to this PR instead?

Don't get me wrong, but this could become a long review otherwise 😉

Reply:
Yes, please do! :-)

Reply:
I can copy it to my local tuya version after and test it to verify it works

Reply:
Done, feel free to test it :)

Reply:
Works perfect!